src/shared/util/dom.tsx

```diff
@@ -12,3 +12,31 @@ export function insertAfter(el: HTMLElement, referenceNode: Node): void {
 export function isMouseEventWithModifierKey(e: MouseEvent): boolean {
     return e.altKey || e.shiftKey || e.ctrlKey || e.metaKey || e.which === 2
 }
+
+export function commitIDFromPermalink({
+    selector,
+    hrefRegex,
+}: {
+    selector: string
+    hrefRegex: RegExp
+}): string | Error {
+    const permalinkElement = document.querySelector(selector)
+    if (!permalinkElement) {
+        return new Error(
+            `Unable to determine the commit ID (40 character hash) you're on because the permalink shortcut element (query selector ${selector}) which contains the commit ID does not exist in the DOM.`
+        )
+    }
+    const href = permalinkElement.getAttribute('href')
```

Comment:
If you accept the above suggestions, you can directly get the `href` property and not need to use `getAttribute`.

```suggestion
    const href = permalinkElement.href
```

Reply:
I would prefer `.href` becuase [I use it elsewhere](https://sourcegraph.com/github.com/sourcegraph/browser-extensions/-/blob/src/libs/gitlab/scrape.ts#L160) but its not worth spending much more time on. I'm fine with `getAtrribute`.